File: SpringBootApp/src/main/java/com/example/springbootapp/Data/DTO/OrderItemDto.java
package com.example.springbootapp.Data.DTO;

import com.example.springbootapp.Data.Entities.Embdeddables.Personalization;
import com.example.springbootapp.Data.Entities.ProductWithVariant;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class OrderItemDto {
    private String id;
    private Integer quantity;
    private Double price;
    private Personalization personalization;
    private ProductWithVariantDto product;
}

File: ClientAdmin/app/src/main/java/com/example/clientadmin/model/dto/OrderItemDto.kt
package com.example.clientadmin.model.dto

import com.example.clientadmin.model.embdeddables.Personalization
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class OrderItemDto (
    val id: String,
    val quantity: Int,
    val price: Double,
    val personalization: Personalization,
    val product: ProductWithVariantDto
)